### PR TITLE
Filling-in optional fields of TransactionRequest...

### PR DESCRIPTION
**Pull request by tomusdrw**
...before sending to Signer UI

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 1a6add8aaf760470eeb10bd719cc2499c483f1fd
  * Your subject line is longer than 72 characters

Guidelines are available at https://github.com/ethcore/parity

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 1a6add8aaf760470eeb10bd719cc2499c483f1fd
  * Your subject line is longer than 72 characters

Guidelines are available at https://github.com/ethcore/parity

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 1a6add8aaf760470eeb10bd719cc2499c483f1fd
  * Your subject line is longer than 72 characters

Guidelines are available at https://github.com/ethcore/parity

---
This message was auto-generated by https://gitcop.com

File: rpc/src/v1/impls/eth_signing.rs [hunk 1]
```diff
@@ -24,25 +24,40 @@ use util::numbers::*;
...
+			miner: Arc::downgrade(miner),
+		}
+	}
+
+	fn fill_optional_fields(&self, miner: Arc<M>, mut request: TransactionRequest) -> TransactionRequest {
+		if let None = request.gas {
+			request.gas = Some(miner.sensible_gas_limit());
+		}
+		if let None = request.gas_price {
+			request.gas_price = Some(miner.sensible_gas_price());
```
Comment (NikVolf) [inline rpc/src/v1/impls/eth_signing.rs]: `trace!` to help with the reasons that request state mutated?

Comment (tomusdrw) [inline rpc/src/v1/impls/eth_signing.rs]: It's added in `../impls/mod.rs` when sending transaction anyhow, it's not really muted it's just filled in (cause dapp developer left this to us to fill reasonable default)

Comment (NikVolf) [inline rpc/src/v1/impls/eth_signing.rs]: well note that value comes from some algorithm and was not set by the user can be helpful, but it's up to you

**Comment by NikVolf** [comment]
`continuous-integration/appveyor/branch` fail can be ignored

File: rpc/src/v1/impls/eth_signing.rs [hunk 2]
```diff
@@ -24,25 +24,40 @@ use util::numbers::*;
...
 	/// Creates a new signing queue client given shared signing queue.
-	pub fn new(queue: &Arc<ConfirmationsQueue>) -> Self {
+	pub fn new(queue: &Arc<ConfirmationsQueue>, miner: &Arc<M>) -> Self {
 		EthSigningQueueClient {
 			queue: Arc::downgrade(queue),
+			miner: Arc::downgrade(miner),
+		}
+	}
+
+	fn fill_optional_fields(&self, miner: Arc<M>, mut request: TransactionRequest) -> TransactionRequest {
```
Comment (rphmeier) [inline rpc/src/v1/impls/eth_signing.rs]: could be worthwhile at some point to make a type-level distinction between requests with fields potentially missing and fully fleshed-out requests.

Comment (tomusdrw) [inline rpc/src/v1/impls/eth_signing.rs]: Yup. In fact it would be best to fill those options in UI, but some of the stuff is not exposed yet. I will create an issue to refactor this part in the (near) future.

**Comment by gavofyork** [comment]
Failed test: v1::tests::mocked::eth_signing::should_add_transaction_to_queue

File: rpc/src/v1/impls/eth_signing.rs [hunk 3]
```diff
@@ -24,25 +24,40 @@ use util::numbers::*;
...
+		if let None = request.gas {
+			request.gas = Some(miner.sensible_gas_limit());
```
Comment (gavofyork) [inline rpc/src/v1/impls/eth_signing.rs]: no - `miner.sensible_gas_limit` is now a fallback and should no longer be used directly. see the implementation of `fn gas_price()` in `eth.rs`.

Comment (tomusdrw) [inline rpc/src/v1/impls/eth_signing.rs]: The logic here is the same as in `impls/mod.rs#sign_and_dispatch`, using signer does not change this behaviour.
Replacing this can go in a separate PR.
